aptos-move/move-examples/marketplace/sources/fee_schedule.move

```diff
@@ -0,0 +1,498 @@
+        let updated_resource = string::utf8(b"fee_address");
+        event::emit_event(&mut fee_schedule_obj.mutation_events, MutationEvent { updated_resource });
+    }
+
+    /// Remove any existing listing fees and set a fixed rate listing fee.
+    public entry fun set_fixed_rate_listing_fee(
+        creator: &signer,
+        marketplace: Object<FeeSchedule>,
+        fee: u64,
+    ) acquires FeeSchedule, FixedRateListingFee {
+        let fee_schedule_signer = remove_listing_fee(creator, marketplace);
+        move_to(&fee_schedule_signer, FixedRateListingFee { listing_fee: fee });
+        let updated_resource = type_info::type_name<FixedRateListingFee>();
+        emit_mutation_event(signer::address_of(&fee_schedule_signer), updated_resource);
+    }
+
+    inline fun remove_listing_fee(
+        creator: &signer,
+        marketplace: Object<FeeSchedule>,
+    ): signer acquires FeeSchedule, FixedRateListingFee {
+        let (fee_schedule_signer, fee_schedule_addr) = assert_access(creator, marketplace);
+        if (exists<FixedRateListingFee>(fee_schedule_addr)) {
+            move_from<FixedRateListingFee>(fee_schedule_addr);
+        };
+        fee_schedule_signer
+    }
+
+    /// Remove any existing bidding fees and set a fixed rate bidding fee.
+    public entry fun set_fixed_rate_bidding_fee(
+        creator: &signer,
+        marketplace: Object<FeeSchedule>,
+        fee: u64,
+    ) acquires FeeSchedule, FixedRateBiddingFee {
+        let fee_schedule_signer = remove_bidding_fee(creator, marketplace);
+        move_to(&fee_schedule_signer, FixedRateBiddingFee { bidding_fee: fee });
+        let updated_resource = type_info::type_name<FixedRateListingFee>();
+        emit_mutation_event(signer::address_of(&fee_schedule_signer), updated_resource);
+    }
+
+    inline fun remove_bidding_fee(
+        creator: &signer,
+        marketplace: Object<FeeSchedule>,
+    ): signer acquires FeeSchedule, FixedRateBiddingFee {
+        let (fee_schedule_signer, fee_schedule_addr) = assert_access(creator, marketplace);
+        if (exists<FixedRateBiddingFee>(fee_schedule_addr)) {
+            move_from<FixedRateBiddingFee>(fee_schedule_addr);
+        };
+        fee_schedule_signer
+    }
+
+    /// Remove any existing commission and set a fixed rate commission.
+    public entry fun set_fixed_rate_commission(
+        creator: &signer,
+        marketplace: Object<FeeSchedule>,
+        commission: u64,
+    ) acquires FeeSchedule, FixedRateCommission, PercentageRateCommission {
+        let fee_schedule_signer = remove_commission(creator, marketplace);
+        move_to(&fee_schedule_signer, FixedRateCommission { commission });
+        let updated_resource = type_info::type_name<FixedRateListingFee>();
+        emit_mutation_event(signer::address_of(&fee_schedule_signer), updated_resource);
+    }
+
+    /// Remove any existing commission and set a percentage rate commission.
+    public entry fun set_percentage_rate_commission(
```

Comment:
This can allow funky behaviors if the marketplace owner changes the commissions after listings are already created when they see that there's a large purchase going through